contracts/RapidCoin.sol

```diff
@@ -234,79 +262,71 @@ contract RapidCoin is StandardToken {
     uint256 saleEndDate;
-
+    address walletForBalanace;
     //TODO presale 4 weeks but full sale start after 4 week
     function RapidCoin () public {
         totalSupply = 10 * (10 ** 9) * (10 ** decimals); //10 billion
         tokenSaleLimit = 475 * (10 ** 7) * (10 ** decimals); //4.75 billions
-        uint256 ownerTokens = totalSupply ; //we assign 10 million to acccountant firstly. 
+        uint256 ownerTokens = totalSupply; //we assign 10 million to acccountant firstly. 
         balances[msg.sender] = ownerTokens;
-        amountRaised = 0;
         owner = msg.sender;
+        walletForBalanace = msg.sender;
         preSaleStartDate = now;//1516840000000; //January 25 2018 00 : 00  UTC
         preSaleEndDate = 1519611199999; // FEB 26 2018 00 : 00 UTC
         saleStartDate = 1516840000000; //January 25 2018 00 : 00  UTC
         saleEndDate = 1515571199000; //January 10 2018 UTC 8 AM and 4 week after presale
     }
-    function () payable whenNotPaused {
-        require(msg.sender !=0x0);
+    function () external payable whenNotPaused {
+       buyTokens();
+    }//end of function callback
+    
+    
+    function buyTokens() public payable whenNotPaused{
+       require(msg.sender != 0x0);
         require(now>=preSaleStartDate);
         require(now<=saleEndDate);
-        uint256 tokens =0;
-        if(now<preSaleEndDate){
+        uint256 tokens = 0;
+       
+        if (now < preSaleEndDate) {
             //presLe handle
              tokens = (msg.value * (10 ** decimals)) / priceOfTokenPREICO;
              uint bonus = 0;
-             if(now < (preSaleStartDate  + 7 * 1  days)){
+             if (now < (preSaleStartDate + 7 days)){
                  bonus = 10;
-             }else if (now < (preSaleStartDate  +14 * 1  days)){
+             }else if (now < (preSaleStartDate + 14 days)) {
                  bonus = 5;
              }
-            uint bonusTokens = (tokens * bonus) /100;
+            uint bonusTokens = (tokens * bonus) / 100;
             tokens += bonusTokens;
             
-        }
-        else if(now>saleStartDate){
+        }else if (now > saleStartDate) {
             //here sale start date
             tokens = (msg.value * (10 ** decimals)) / priceOfToken;
-        }//end of else if for sale start date
-        else {
+        }else {
             revert(); //this revert is between pre sale and sale time gap
         }
-        allowed[owner][msg.sender]+=tokens;
-        bool result =  transferFrom(owner,msg.sender,tokens);
-    }//end of function callback
-    
-    //function ext
-
-    /**
-    * Transfer entire balance to any account (by owner and admin only)
-    **/
-    function transferFundToAccount(address _accountByOwner) public onlyOwner {
-        require(amountRaised > 0);
-        _accountByOwner.transfer(amountRaised);
-    }
+        allowed[owner][msg.sender] += tokens;
+        bool result = transferFrom(owner,msg.sender,tokens);
+        if (!result) {
```

Comment:
@yasiriqbal776 could be replaced by require(result); Actually, there is no need for checking transferFrom methods. It will throw an exception of something will go wrong. So if (!result) could be removed anyway.